var/spack/repos/builtin/packages/nest-simulator/package.py

```diff
@@ -0,0 +1,138 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class NestSimulator(CMakePackage):
+    """NEST is a simulator for spiking neural network models
+
+    It focuses on the dynamics, size and structure of neural systems rather
+    than on the exact morphology of individual neurons."""
+
+    homepage = "http://www.nest-simulator.org"
+    url      = "https://github.com/nest/nest-simulator/releases/download/v2.12.0/nest-2.12.0.tar.gz"
+
+    version('2.14.0', sha256='d6316d6c9153100a3220488abfa738958c4b65bf2622bd15540e4aa81e79f17f')
+    version('2.12.0', sha256='bac578f38bb0621618ee9d5f2f1febfee60cddc000ff32e51a5f5470bb3df40d')
+    version('2.10.0', sha256='2b6fc562cd6362e812d94bb742562a5a685fb1c7e08403765dbe123d59b0996c')
+    version('2.8.0',  sha256='d47325b27a5599b6ea58a3c4ef06656e7c5a4941c4e94dec6a5c2fa956209915')
+    version('2.6.0',  sha256='5fe4924bc57d0c7dd820aa371de935eedf7e813832c0eee2c976b33c9a8db4cf')
+    version('2.4.2',  sha256='8f86e58c1a12b733ffabd8b0400326e5a3494a458149ea8ebe9f19674d05b91b')
+
+    variant('python', default=True,
+            description='Build the PyNest interface')
+    variant('mpi', default=True,
+            description='Build with MPI bindings')
+    variant('optimize', default=True,
+            description='Build with MPI bindings')
+    variant('modules', default=False,
+            description='Enables external module support')
+    # TODO add variants for neurosim and music when these are in spack
+
+    depends_on('python',            when='+python')
+    depends_on('py-numpy',          when='+python')
+    depends_on('py-cython@0.19.2:', when='+python')
+    depends_on('py-nose',           when='+python')
+    depends_on('py-setuptools',     when='+python')
```

Comment:
`python` and `py-numpy` are probably `type=('build', 'run')`, `py-cython` and `py-setuptools` are probably `type='build'`, and `py-nose` is probably `type='test'`.